pybio/spec/pybio_types.py

```diff
@@ -0,0 +1,167 @@
+from dataclasses import dataclass
```

Comment:
We'll require Python 3.7 when using dataclasses, right?

Reply:
yes, is that a problem?

Reply:
Not specifically I think. It depends on what environments we want to support and how easy to make it for users.

- Ubuntu 18.04 still defaults to Python 3.6. Those users will  need to install Python 3.7 eg from a ppa.
- Debian Buster comes with Python 3.7.
- OS X doesn't come with Python 3 at all, as far as I know, so it doesn't matter here.
- Windows users will also need to install Python manually.
- Google colab only supports Python 3.6 so far.

Do we see any limitations with the above?

Reply:
- Ubuntu 20.04 will have Python 3.8.

Reply:
@MartinHjelmare thanks for bringing this up, these are valid concerns.
We are a bit biased here by having all our workflows on conda, which makes switching to newer python versions relatively easy.
However, loosing compatibility with the default ubuntu python version and the default for colab is certainly something to consider, because there might be a potential user base who uses these set ups with system python.
So I think it depends on whether we want to limit support to conda. If this is the case, we can safely go with python 3.7. If we want to support system python + pip though, support for python 3.6 would be desirable.

Reply:
I like using newer Python versions too.

@oeway does it matter from your perspective?

Reply:
I think it would be great if we make it compatible with python 3.6, considering this going to be a shared library. For BioEngine for example, we would like to run it in google colab.

Reply:
There seems to be a backport of dataclasses specifically for Python 3.6. We'd have to install it only on Python 3.6 I guess.
https://github.com/ericvsmith/dataclasses